pkg/provisioner/ironic/factory_test.go

```diff
@@ -181,35 +181,78 @@ func TestLoadConfigFromEnv(t *testing.T) {
 	}
 }
 
-func TestLoadEndpointsFromEnv(t *testing.T) {
-	cases := []struct {
-		name        string
-		env         EnvFixture
-		expectError bool
-	}{
-		{
-			name: "with-ironic",
-			env: EnvFixture{
-				ironicEndpoint: "http://ironic.test",
-			},
-		}, {
-			name:        "without-ironic",
-			env:         EnvFixture{},
-			expectError: true,
-		},
+func TestLoadTLSConfigFromEnv(t *testing.T) {
+	const (
+		TLSKeyFilePath  = "/opt/metal3/certs/client/tls.key"
+		TLSCertFilePath = "/opt/metal3/certs/client/tls.crt"
+	)
+
+	os.Setenv("IRONIC_CACERT_FILE", "/path/to/ca.crt")
+	os.Setenv("IRONIC_CLIENT_CERT_FILE", "/path/to/client.crt")
+	os.Setenv("IRONIC_CLIENT_PRIVATE_KEY_FILE", "/path/to/client.key")
+	os.Setenv("IRONIC_INSECURE", "false")
+	os.Setenv("IRONIC_SKIP_CLIENT_SAN_VERIFY", "true")
+	defer func() {
+		os.Unsetenv("IRONIC_CACERT_FILE")
+		os.Unsetenv("IRONIC_CLIENT_CERT_FILE")
+		os.Unsetenv("IRONIC_CLIENT_PRIVATE_KEY_FILE")
+		os.Unsetenv("IRONIC_INSECURE")
+		os.Unsetenv("IRONIC_SKIP_CLIENT_SAN_VERIFY")
+	}()
```

Comment:
If you check the other tests in this file, you will see that they make use of a struct [EnvFixture](https://github.com/metal3-io/baremetal-operator/blob/aef07f5593ae48d589275d88cd43015c9171d7e7/pkg/provisioner/ironic/factory_test.go#L10-L18). There are [helpful functions](https://github.com/metal3-io/baremetal-operator/blob/aef07f5593ae48d589275d88cd43015c9171d7e7/pkg/provisioner/ironic/factory_test.go#L20-L46) for setting and unsetting the environment from this struct. Put all of this together and you get a nice way of looping through test cases described by structs. Could you try to mimic this?